### PR TITLE
#3939 Acceptable boolean syntax differs for data writes and data queries

### PR DESCRIPTION
**Pull request by apriendeau**
So I made the assumption that `T` and `F` are acceptable boolean values but I am not sure where the stance is on supporting that including all the variations of the `true/false` but if that is the case and we do decide to go forward with this, there are tests to cover all the variations of `true/false`. So I guess, I am asking which direction we would like to go.

### CODE REVIEW

**Comment by otoolep** [comment]
Thanks @apriendeau -- have you signed the CLA?

Also, the commit messages are not very clear. Can you squash the two commits, and include a clearer commit message?

File: CHANGELOG.md [hunk 1]
```diff
@@ -10,6 +10,7 @@
 - [#4111](https://github.com/influxdb/influxdb/pull/4111): Update pre-commit hook for go vet composites
 - [#4136](https://github.com/influxdb/influxdb/pull/4136): Return an error-on-write if target retention policy does not exist. Thanks for the report @ymettier
 - [#4124](https://github.com/influxdb/influxdb/issues/4124): Missing defer/recover/panic idiom in HTTPD service
+- [#3939](https://github.com/influxdb/influxdb/issues/3939): [0.9.3] Acceptable boolean syntax differs for data writes and data queries
```
Comment (otoolep) [inline CHANGELOG.md]: Please give yourself credit here `Thanks @apriendeau` is what we like to write.

**Comment by otoolep** [comment]
I like the idea of symmetry, so I am OK with this. +1

@dgnorton

File: CHANGELOG.md (hunk 1 above)
Comment (apriendeau) [inline CHANGELOG.md]: you want me to thank myself?

Comment (otoolep) [inline CHANGELOG.md]: If you don't, we will. :-)

Comment (apriendeau) [inline CHANGELOG.md]: okay, I appreciate it.

**Comment by apriendeau** [comment]
Yes. I can squash them I was confused and I have signed the CLA.

**Comment by apriendeau** [comment]
@otoolep not squashed because it was a merge conflict but reworded to be better.

**Comment by dgnorton** [comment]
I don't like having `t` and `f` as keywords in the query language.  In the write path, single character bools save network traffic and parsing time, which will add up when processing billions of points.  Most of those points are machine generated and never looked at by humans.  Queries are low volume by comparison and often looked at by humans.  Let's get a few more opinions.  I may get out voted.  :)  @pauldix @benbjohnson @corylanou @beckettsean @DanielMorsing ?

**Comment by otoolep** [comment]
I don't have a strong opinion, you raise a fair point @dgnorton

**Comment by benbjohnson** [comment]
I'm not a fan of `t` and `f` as keywords either. I'd rather we limit keywords as much as possible.

**Comment by apriendeau** [comment]
I am not a fan of 't/f' as keywords but isn't it implied when you can use it in a write? So if we remove it, I would suggest that we only have `true/false` because having it on one side or the other, implies that it is a keyword. Currently we have:

Boolean syntax |  Writes | Queries  |
------------------------|-----------|--------------|
 `t`,`f` |	&#10003; | ❌ |
 `T`,`F` |  &#10003; |  ❌ |
 `true`,`false` | &#10003; | &#10003; |
  `True`,`False` |  &#10003; |  &#10003; |
 `TRUE`,`FALSE` | &#10003; |  &#10003; |

I am in favor of not supporting `t/f` but I do think the best syntax to not support it for writes either and do a conversion under the hood for writes so that you are always getting the performance.

**Comment by toddboom** [comment]
@apriendeau i think it might be good to also have some coverage of this in an integration test to demonstrate that it's correct all the way through the system. also, i think @corylanou's suggestion to do #4037 alongside this is a good idea.

**Comment by benbjohnson** [comment]
@apriendeau That's a good point. I think I'd personally rather see `t`/`f` removed from writes rather than added to the query language.

**Comment by apriendeau** [comment]
@toddboom Sure, I can would be more than happy to add #4037 that to PR while we are discussing the syntax :+1:

**Comment by apriendeau** [comment]
@toddboom @corylanou Which way would you guys lean towards for acceptable write/read for booleans? If we are looking towards the other route, which is to remove support for t/f in the write query, I  will probably close this PR and opened a new one.

**Pull request closed without merging**